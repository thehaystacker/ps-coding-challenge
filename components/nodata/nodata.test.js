import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Nodata from "./nodata";

describe("Nodata", () => {
  test("should render the component without crashing", () => {
    const { getByTestId } = render(<Nodata />);
    const message = getByTestId("message");

    expect(message).toBeVisible();
  });

  test("should output default message", () => {
    const { getByText } = render(<Nodata />);
    const message = getByText(`Oops! No data found`);

    expect(message).not.toBeNull();
  });

  test("should output the message passed as prop", () => {
    const { getByText } = render(<Nodata message="test message" />);
    const message = getByText(`test message`);

    expect(message).not.toBeNull();
  });

  test("should match the snapshot", () => {
    const { asFragment } = render(<Nodata />);
    expect(asFragment()).toMatchSnapshot();
  });
});
