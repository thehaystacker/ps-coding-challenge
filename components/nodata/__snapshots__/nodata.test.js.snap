// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nodata should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ps-card"
  >
    <div
      class="wrapnodata"
    >
      <h4
        data-testid="message"
      >
        Oops! No data found
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
